src/io-methods/filesystem.js

```diff
@@ -1,10 +1,19 @@
 "use strict";
 const fs = require("fs/promises");
 const { getConfig } = require("../config");
+const { getEmptyManifest } = require("../modify");
 const jsHelpers = require("./js-file-helpers.js");
 
-exports.readManifest = function (filePath) {
-  return fs.readFile(filePath, "utf-8");
+exports.readManifest = async function (filePath) {
+  try {
+    await fs.access(filePath, fs.F_OK);
+
+    return fs.readFile(filePath, "utf-8");
+  } catch (missingFileErr) {
+    exports.writeManifest(filePath, JSON.stringify(getEmptyManifest()));
```

Comment:
I think we need to await this in order to avoid race condition between the writing of empty manifest and reading of it afterwards:

```suggestion
    await exports.writeManifest(filePath, JSON.stringify(getEmptyManifest()));
```

Reply:
Good catch!